src/main/java/com/untamedears/JukeAlert/command/commands/InfoCommand.java

```diff
@@ -12,33 +13,63 @@
 import org.bukkit.command.CommandSender;
 import org.bukkit.entity.Player;
 
-import vg.civcraft.mc.civmodcore.command.PlayerCommand;
-import vg.civcraft.mc.namelayer.permission.PermissionType;
-
 import com.untamedears.JukeAlert.JukeAlert;
+import com.untamedears.JukeAlert.model.LoggedAction;
 import com.untamedears.JukeAlert.model.Snitch;
 import com.untamedears.JukeAlert.tasks.GetSnitchInfoPlayerTask;
 
+import vg.civcraft.mc.civmodcore.command.PlayerCommand;
+import vg.civcraft.mc.namelayer.permission.PermissionType;
+
 public class InfoCommand extends PlayerCommand {
 
     public class History {
-        public History(int snitchId, int page) {
+        public History(int snitchId, int page, LoggedAction filterAction, String filterPlayer) {
             this.snitchId = snitchId;
             this.page = page;
+            this.filterAction = filterAction;
+            this.filterPlayer = filterPlayer;
         }
         public int snitchId;
         public int page;
+        public LoggedAction filterAction;
+        public String filterPlayer;
     }
-
+    
     private static Map<UUID, History> playerPage_ = new TreeMap<UUID, History>();
+    private static final String[] autocompleteCommands = {"next", "censor", "action=", "player="};
 
     public InfoCommand() {
         super("Info");
         setDescription("Displays information from a Snitch");
-        setUsage("/jainfo <page number or 'next'> [censor]");
-        setArguments(0, 2);
+        setUsage("/jainfo [<page number> or 'next'] [censor] [action=<action type>] [player=<player name>]");
+        setArguments(0, 6); // Max args = 6 because the action might be split into two
         setIdentifier("jainfo");
     }
+    
+    /**
+     * Attempts to match a string to one of the LoggedAction actions
+     * @param action - case insensitive
+     * @return The matching LoggedAction, or null if a matching action couldn't be found
+     */
+    private static LoggedAction decodeAction(String action){
```

Comment:
So, LoggedAction is an enumeration iirc. Simple add an overridden "valueOf" or a "decode" method there, and resolve String-to-Enum within the Enumeration, not as a random static function on the command handler.

Then when you want to get the enum, just LoggedAction.valueOf(string);